### ISSUE TITLE
AV software blocking dbatools import - flagging as malicious

### ISSUE THREAD

**Issue by jemmiegod** [dataplat/dbatools#8241]
### Verified issue does not already exist?

Yes

### What error did you receive?

ParserError: C:\...\PowerShell\Modules\dbatools\1.1.80\allcommands.ps1:1
Line |
   1 |  ### DO NOT EDIT THIS FILE DIRECTLY ###
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | This script contains malicious content and has been blocked by your antivirus software.

### Steps to Reproduce

- Open a new PowerShell terminal
- Install-Module -MinimumVersion 1.1.80 -Name dbatools
- Import-Module dbatools

### Are you running the latest release?

Yes

### Other details or mentions

There was a discussion a while back about this, but the issue seems to have reappeared
https://github.com/PowerShell/PowerShell/issues/15396

I have tried multiple versions of dbatools without success.  Versions that have previously worked for months, no longer work
I have submitted this as a false positive to FireEye already.

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe), Windows PowerShell (powershell.exe), Windows PowerShell ISE (powershell_ise.exe), VS Code (terminal), VS Code (integrated terminal)

### PowerShell Host Version

Name                           Value
----                           -----
PSVersion                      7.1.5
PSEdition                      Core
GitCommitId                    7.1.5
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

### SQL Server Edition and Build number

Microsoft SQL Server 2019 (RTM) - 15.0.2000.5 (X64)   Sep 24 2019 13:48:23   Copyright (C) 2019 Microsoft Corporation  Express Edition (64-bit) on Windows 10 Enterprise 10.0 <X64> (Build 19042: ) (Hypervisor) 

### .NET Framework Version

.NET 5.0.11

**Comment by wsmelton** [comment]
I don't believe there is much we can do about this. There are certain issues with Defender as well flagging it too that I don't think have been fully fixed yet.

**Comment by andreasjordan** [comment]
We shipped some updates so this is probably not the case anymore. So I will close this.

**Comment by Brett-Jay** [comment]
I am getting this error using DBATOOLS version 1.1.103. We use Carbon Black. DBATOOLS was working up until about 3 weeks past. Now we get this error when importing the module and Carbon Black pops a message that a Deny Action was applied. 

PS C:\Users\xxxxxxx> Import-Module Dbatools
 
At C:\Program Files\WindowsPowerShell\Modules\Dbatools\1.1.103\allcommands.ps1:1 char:1
+ ### DO NOT EDIT THIS FILE DIRECTLY ###
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
+ CategoryInfo : ParserError: (:) [], ParseException
+ FullyQualifiedErrorId : ScriptContainedMaliciousContent

**Comment by nicpenning** [comment]
Same issue here. This is being flagged as malicious by Carbon Black today as well. 1.1.105 is being blocked with the same message as above. Reopen this issue? Or create a new one?

**Comment by andreasjordan** [comment]
Neither will have an effect - as we (the developers) can do nothing about this. If you are able to work with the vendor of Carbon Black on the details (like what specific file is maybe not signed), then we might be able to change something to prevent this.

**Comment by wsmelton** [comment]
Overall those are false positives as they are not malicious. If they are indeed flagging for that value @nicpenning then it comes from this:

https://github.com/dataplat/dbatools/blob/6cae0dd18bda3ad8efd60404c2d05b402cc4a785/functions/Get-DbaWaitStatistic.ps1#L493-L500

**Comment by nicpenning** [comment]
Of course. Nothing you can really do to fix that. Also, like I mentioned, this was just one instance of CB flagging the code. Even if you got rid of that part of the code, there are other "red flag" text that get tripped but I don't have the time to find them. VMWare will need to adjust the detection/prevention mechanisms.

**Comment by potatoqualitee** [comment]
yeah, please report to CB. there's no way for us to contact them that I'm aware of :/

**Comment by Brett-Jay** [comment]
We have entered a support request with Carbon Black.

**Comment by potatoqualitee** [comment]
Thank you! Please let us know how it goes

**Comment by potatoqualitee** [comment]
Reopening as it's continuing to happen to others.

**Comment by Brett-Jay** [comment]
CB came back and said that in the next iterations of the main agent software, they will integrate a fix. If not the next upcoming version, it'll be in the one afterward.

**Comment by potatoqualitee** [comment]
Oh, awesome, thank you @Brett-Jay !

**Comment by Geo-Ron** [comment]
Issue seems related to PSFramework integration: https://github.com/PowershellFrameworkCollective/psframework/issues/517
We are currently working on resolving this with CB support

**Comment by wsmelton** [comment]
@Geo-Ron PSFramework is not implemented in dbatools. The module originated in a way from work Fred did with our messaging system but I don't believe it is can be considered the se thing anymore.

**Comment by potatoqualitee** [comment]
yeah, CB is just hating on dbatools, Pester and more. They really need to fix old issue and are dragging 🍑 , repeatedly promising "next release" and apparently not following through.

**Comment by potatoqualitee** [comment]
@Geo-Ron please keep us updated. they also dont accept reports from non-customers so there's literally nothing we can do about it 😡

**Comment by wsmelton** [comment]
Going to start using this as a tracker for AV software causing PowerShell issues, particularly with our module. Be aware that there is nothing we (as maintainers) can do about this issue. The AV software in general is generally going to flag multiple modules similar to ours as being malicious. 

Please 👍 the post according to your AV software, if you don't see it listed in this issue please add a post with the AV name. If anyone has links on how to report incorrect findings to these AV vendors please share those too.

FireEye - https://github.com/dataplat/dbatools/issues/8241#issue-1170200178
Carbon Black - https://github.com/dataplat/dbatools/issues/8241#issuecomment-1147959896

**Comment by wsmelton** [comment]
#8949 - blocking by CB occurring with v2 release.

**Comment by andreasjordan** [comment]
@wsmelton What label do you suggest here? Or should we close as duplicate?

**Comment by wsmelton** [comment]
Doesn't need a label IMO.

**Comment by tomaustin700** [comment]
Started seeing this today after never seeing it ever before. We're just running Windows Defender.

**Comment by hiralm21** [comment]
This happened with us last night too. We're just running Windows Defender.

**Comment by niphlod** [comment]
seems to be tied to the latest update 

![Image](https://github.com/user-attachments/assets/8d65df4f-4429-49d1-9aea-bf3a7aab7bc5)

**Comment by flagmanchris** [comment]
Re Defender blocking it (first spotted on our estate 09/06/2025 17:00) we reported it to MS this morning; they verified false positive and advised will be resolved in security intelligence update. They didn't give a timeline for the update though

**Comment by jakefjord** [comment]
> Re Defender blocking it (first spotted on our estate 09/06/2025 17:00) we reported it to MS this morning; they verified false positive and advised will be resolved in security intelligence update. They didn't give a timeline for the update though

Do you have any expectation for how long before a fix is implemented? We have many automated systems that depend on this. Should we be re-tooling those or wait it out?

**Comment by JeffersonGhUsr** [comment]
Seems to be resolved with latest MS Defender security intelligence update version 1.429.483.0 !

**Comment by flagmanchris** [comment]
Yes, I've just tested and seen the same. Not had confirmation back from MS but it does look to be resolved

**Comment by flagmanchris** [comment]
Just had confirmation from MS that it is resolved in 1.429.483.0 as we've seen

**Comment by BrianL-STCU** [comment]
My updates today seem to have already fixed the issue, likely [KB5060842 Win11 OS update](https://support.microsoft.com/en-us/topic/june-10-2025-kb5060842-os-build-26100-4349-47ff300b-2a04-440c-9476-2860d04fce8d).

**Comment by potatoqualitee** [comment]
Thanks all, going to close this one bc they do this to us sometimes. if you can open new tickets so they can be bumped to the top and be more visible, we'd appreciate it.

**Comment by potatoqualitee** [comment]
Please try with the latest version. dbatools is now signed with a Microsoft-issued certificate which has immediate reputation and I don't think we'll have anymore AV issues. You can read more here: https://blog.netnerds.net/2025/08/dbatools-azure-trusted-signing/
